### PR TITLE
Clarify refetch and useLazyQuery

### PR DESCRIPTION
**Pull request by StephenBarlow**

### CODE REVIEW

File: docs/shared/query-result.mdx [hunk 1]
```diff
@@ -183,7 +183,7 @@ This field is only present on the result object returned by [`useLazyQuery`](htt
 
 A function that enables you to re-execute the query, optionally passing in new `variables`.
 
-The `fetchPolicy` for the refetched query will be set to `network-only`, unless the `fetchPolicy` is already `no-cache` or `cache-and-network` (which also guarantee a network request).
+To guarantee that the refetch performs a network request, its `fetchPolicy` is set to `network-only` (unless the original query's `fetchPolicy` is `no-cache` or `cache-and-network`, which also guarantee a network request).
```
Comment (brainkim) [inline docs/shared/query-result.mdx:186]: @benjamn don’t we have logic which forces a `fetchPolicy` of `network-only` when refetch is called?

Comment (benjamn) [reply docs/shared/query-result.mdx:186]: @brainkim I believe @StephenBarlow's new text is accurate, given this implementation?
https://github.com/apollographql/apollo-client/blob/a3f8e99920bb6aa05d85207143e79eccb0023fb4/src/core/ObservableQuery.ts#L304-L312

**Review by stemmlerjs — APPROVED** [review]
This helps a lot. Thanks @StephenBarlow!

**Comment by benjamn** [comment]
@stemmlerjs @StephenBarlow Here's a new PR that warns when `refetch({ variables })` is called (and the query doesn't declare a `$variables` variable), which seems related to these clarifications: #8702

**Pull request merged**